### PR TITLE
[ENH] dedicated `_update` for `BATS` and `TBATS`

### LINKED ISSUE #3063: [ENH] Implement update method for TBATS
**Is your feature request related to a problem? Please describe.**
Want to perform walk-forward predictions using TBATS model. without refitting parameters every time. This is crucial for validation, especially since TBATS model is slow to train. 

**Describe the solution you'd like**
Implement update method for TBATS... same as interface used for other time series forecasters in sktime.

### PR DESCRIPTION
**Pull request by jelc53**
…r BATS and TBATS wrappers

#### Reference Issues/PRs
Fixes #3063. See also https://github.com/intive-DataScience/tbats/issues/35 for discussion. 

#### What does this implement/fix? Explain your changes.
Implements update method for tbats, which allows users to update state fo the model with new observations (fast) without refitting model parameters (slow). Particualrly, useful for walk-forward prediction and validation.

#### Does your contribution introduce a new dependency? If yes, which one?
No new dependencies were introduced.

#### What should a reviewer concentrate their feedback on?
There are a number of potential places I could implement this change. I thought the _tbats.py file with the adapter class was the most natural but let me know if this was not the intended design / pattern. 

#### Any other comments?
This is my first contribution to a public code base. Would appreciate any feedback and, if happy, would be interested in contributing more (eg. related issues #3064 and #3065).

### CODE REVIEW

File: sktime/forecasting/base/adapters/_tbats.py [hunk 1]
```diff
@@ -102,10 +102,39 @@ def _fit(self, y, X=None, fh=None):
...
+        X : pd.DataFrame, optional (default=None)
+            Exogenous variables (ignored)
+        update_params : bool, optional (default=True)
+            whether model parameters should be updated
 
+        Returns
+        -------
+        self : reference to self
+        """
+        if update_params:
```
Comment (fkiraly) [inline sktime/forecasting/base/adapters/_tbats.py:125]: I think instead this should be as follows:

`if update_params`, then run `self._forecaster.fit`.

Else, *only return self*.

Why: `update_params=False` means data is ingested, but the internal models are *unchanged*.

**Review by fkiraly — CHANGES_REQUESTED** [review]
Great, thanks!

Two change requests:
* code formatting (linting) fails, kindly check "details" with the first CI element. It is also a good idea to set up linting checks locally.
* `update_params=False` mode should not change any self attributes. This mode is used when only data is ingested, but nothing whatsoever is updated.

File: sktime/forecasting/base/adapters/_tbats.py (hunk 1 above)
Comment (jelc53) [reply sktime/forecasting/base/adapters/_tbats.py:125]: I think this is what happens by default (update model state only, not params) with the TBATS fit method if the model is not re-instantiated. 

A bit weird I know, so maybe I am misinterpretting. See example script here: https://github.com/intive-DataScience/tbats/blob/master/examples/re_fit_model.py. I was able to reproduce something similar for my own project.

Comment (jelc53) [reply sktime/forecasting/base/adapters/_tbats.py:125]: When ``update_params=False`` my understanding is that we want to change two things only: (a) self._y, (b) cutoff index within TBATS model class. We handle (a) I think already in the ``BaseForecaster`` class ``update`` method by calling ``_update_y_X`` but I think need to call TBAT's fit method for this to fix the cuttoff such that we can access other TBATS methods like forecast, etc. after the data update.

Comment (fkiraly) [reply sktime/forecasting/base/adapters/_tbats.py:125]: I see, thanks for explaining!

Then this is ok.

**Comment by fkiraly** [comment]
linting still failing, please click here to see what´s wrong:

image

**Comment by jelc53** [comment]
Oh gotcha, looks like still a trailing white space issue. Will fix when back at my laptop.

**Comment by fkiraly** [comment]
Linting still failing 😁 

Consider it a fun scavenger hunt 😄

**Comment by jelc53** [comment]
Damn, aplogoies for the back and forth. I've tried again (shifting comments to not be end-of-line, found whitespace on an unused line between methods), but not sure how to find and test the linting error systematically without workflow run permissions. Let me know if there is a way I can easily recreate linting requirements locally if it fails again.

**Comment by fkiraly** [comment]
>Let me know if there is a way I can easily recreate linting requirements locally if it fails again.

Sure! The easiest way is installing precommit:
https://www.sktime.org/en/stable/developer_guide/continuous_integration.html

If you want to avoid automated changes to your files, you can alternatively install `black` and `flake8` and activate them in your developer IDE (e.g., in vs code you need to enable them in preferences) to get problems listed.

**Comment by fkiraly** [comment]
I expanded the docs on setting up local code quality checks, feel free to look into the expanded version of the docs:
https://github.com/alan-turing-institute/sktime/pull/3089

**Comment by jelc53** [comment]
Setup ``black`` and ``flake8`` on vscode locally, plus installed ``pre-commit`` so should have the linting part resolved
image

**Comment by fkiraly** [comment]
>Setup ``black`` and ``flake8`` on vscode locally, plus installed ``pre-commit`` so should have the linting part resolved

Indeed! Now we can look at the "serious" tests 😁 

Would also appreciated feedback what you think about the improved documentation around linting in #3089 that I wrote on this occasion. Is that more clear and helpful? Can it be improved?

**Review by fkiraly — APPROVED** [review]
Nice! Tests pass! Thanks!

**Comment by jelc53** [comment]
Great to hear! Appreciate you helping me through it. Re feedback on #3089, the pre-commit setup made life much easier for me so I think that’ll be helpful for others too. Only question for me is whether you also make clear that another option is to configure the ide to flag black and flake8 conventions for those who don’t want to use pre-commit. Not sure. Might be one of those things where more info is more confusing than helpful.

**Pull request merged**